src/lib/Components/Bidding/Screens/ConfirmFirstTimeBid.tsx

```diff
@@ -21,29 +22,73 @@ import { Title } from "../Components/Title"
 
 import SwitchBoard from "lib/NativeModules/SwitchBoard"
 
-import { ConfirmBid_sale_artwork } from "__generated__/ConfirmBid_sale_artwork.graphql"
 import { Checkbox } from "../Components/Checkbox"
+import { Timer } from "../Components/Timer"
+import { BillingAddress } from "./BillingAddress"
+import { ConfirmBidProps } from "./ConfirmBid"
+
+export interface Address {
+  fullName: string
+  addressLine1: string
+  addressLine2?: string
+  city: string
+  state: string
+  postalCode: string
+}
 
-interface ConfirmBidProps extends ViewProperties {
-  sale_artwork: ConfirmBid_sale_artwork
-  bid: {
-    display: string
-    cents: number
-  }
+interface ConformBidState {
+  billingAddress?: Address
+  creditCardToken?: string // TODO: change this interface accrodingly when adapting stripe
+  conditionsOfSaleChecked: boolean
+  isLoading: boolean
 }
 
-export class ConfirmFirstTimeBid extends React.Component<ConfirmBidProps> {
+export class ConfirmFirstTimeBid extends React.Component<ConfirmBidProps, ConformBidState> {
+  state = {
+    billingAddress: undefined,
+    creditCardToken: null,
+    conditionsOfSaleChecked: false,
+    isLoading: false,
+  }
+
   onPressConditionsOfSale = () => {
     SwitchBoard.presentModalViewController(this, "/conditions-of-sale?present_modally=true")
   }
+
+  showCreditCardForm() {
+    // TODO: implement this function: https://artsyproduct.atlassian.net/browse/PURCHASE-89
+    console.warn("The function showCreditCardForm() is not implemented yet.")
+  }
+
+  showBillingAddressForm() {
+    this.props.navigator.push({
+      component: BillingAddress,
+      title: "",
+      passProps: {
+        onSubmit: this.onBillingAddressAdded,
+        billingAddress: this.state.billingAddress,
+      },
+    })
+  }
+
+  onBillingAddressAdded = (values: Address) => {
+    this.props.navigator.pop()
```

Comment:
This is a bit interesting. In iOS apps generally, the child view controller is the one who is responsible for popping like this (instead of in the parent's delegate callback). I've checked in Emission and we've followed that convention here, too ([see this example](https://github.com/artsy/emission/blob/38b7220d944754dc5940ecfed604aa4afcafb5b7/src/lib/Components/Consignments/Screens/Provenance.tsx#L28-L31)).

So what I'd recommend doing here is moving `this.props.navigator.pop()` to the end of `onSubmit()` in `BillingAddress` instead.